ui/src/shared/components/MultiSelectDropdown.js

```diff
@@ -0,0 +1,123 @@
+import React, {Component, PropTypes} from 'react'
+import OnClickOutside from 'shared/components/OnClickOutside'
+import classNames from 'classnames'
+import _ from 'lodash'
+
+class MultiSelectDropdown extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isOpen: false,
+      localSelectedItems: this.props.selectedItems,
+    }
+
+    this.onSelect = ::this.onSelect
+    this.onApplyFunctions = ::this.onApplyFunctions
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (!_.isEqual(this.state.localSelectedItems, nextProps.selectedItems)) {
+      this.setState({
+        localSelectedItems: nextProps.selectedItems,
+      })
+    }
+  }
+
+  handleClickOutside() {
+    this.setState({isOpen: false})
+  }
+
+  toggleMenu(e) {
+    e.stopPropagation()
+    this.setState({isOpen: !this.state.isOpen})
+  }
+
+  onSelect(item, e) {
+    e.stopPropagation()
+
+    const {localSelectedItems} = this.state
+
+    let nextItems
+    if (this.isSelected(item)) {
+      nextItems = localSelectedItems.filter((i) => i !== item)
+    } else {
+      nextItems = localSelectedItems.concat(item)
+    }
+
+    this.setState({localSelectedItems: nextItems})
+  }
+
+  isSelected(item) {
+    return this.state.localSelectedItems.indexOf(item) > -1
+  }
+
+  onApplyFunctions(e) {
+    e.stopPropagation()
+
+    this.setState({isOpen: false})
+    this.props.onApply(this.state.localSelectedItems)
+  }
+
+  render() {
+    const {localSelectedItems, isOpen} = this.state
+    const {label} = this.props
+    const labelText = isOpen ? "0 Selected" : "Apply Function"
+
+    return (
+      <div className={classNames('dropdown multi-select-dropdown', {open: isOpen})}>
+        <div onClick={::this.toggleMenu} className="btn btn-xs btn-info dropdown-toggle" type="button">
+          <span className="multi-select-dropdown__label">
+            {
+              label ?
```

Comment:
Your linter should be yelling at you about nested ternary expressions. I'm cool with them but wonder if this indicates a useful refactor?